blockstm/txnrunner.go

```diff
@@ -0,0 +1,227 @@
+package blockstm
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+
+	"cosmossdk.io/collections"
+	storetypes "cosmossdk.io/store/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+type (
+	DeliverTxFunc func(tx []byte, ms storetypes.MultiStore, txIndex int, incarnationCache map[string]any) *abci.ExecTxResult
+	TxRunner      interface {
+		Run(context.Context, storetypes.MultiStore, [][]byte, DeliverTxFunc) ([]*abci.ExecTxResult, error)
```

Comment:
are these functions defined in the sdk types?

Reply:
Not at the moment. We can do.